app/src/main/java/com/canvearth/canvearth/server/FBPixelManager.java

```diff
@@ -119,10 +119,16 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
         return fbPixel;
     }
 
+    public void getBitmapAsync(PixelData pixelData, int resolutionFactor, Function<Bitmap> callback) {
+        new Thread(()->{
+            callback.run(getBitmapSync(pixelData, resolutionFactor));
+        }).start();
+    }
+
     // You don't have to watch this pixel (for now).. I'm nervous about performance issue of this method.
     // returns Bitmap which has resolution of 2^resolutionFactor * 2^resolutionFactor
     // TODO cache this when there is performance issue
-    // Do we need Async version of this?
+    // TODO this seems better to executed in server side.
```

Comment:
What does this mean?

Reply:
In this implementation, this recursive update logic is executed at each device.
Since I don't have good sense of performance of devices, I am nervous of using devices' resource a lot.
This recursive logic can be executed at the server side(such as aws lambda), so when we have to optimize devices' resources, we can save this execution.

Reply:
Seems plausible. Maybe AWS Lambda can return the endpoint for the image, so it can be used like https://github.com/googlemaps/android-samples/blob/master/ApiDemos/java/app/src/main/java/com/example/mapdemo/TileOverlayDemoActivity.java. But we will have to see the performance on a few devices first.